packages/core/src/utils/ValidJWEStucture.ts

```diff
@@ -0,0 +1,5 @@
+import type { EncryptedMessage } from '../types'
+
+export function validJWEStucture(message: any): message is EncryptedMessage {
```

Comment:
This would allow it to accept a string, but I think it might be better to send the parsed object to this function.

```suggestion
export function validJWEStucture(message: string): boolean {
  const parsedMessage = JSON.parse(message)
  return parsedMessage.protected && parsedMessage.iv && parsedMessage.ciphertext && parsedMessage.tag
}
```

Reply:
@TimoGlastra

Reply:
I also think that we should use `JsonEncoder.fromString()` and not `JSON.parse()`. Right now, `JsonEncoder.fromString()` just wraps `JSON.parse()`, but it might be nice to expand this function in the future to take a generic and do a type comparison (if the conversion went correctly).

Reply:
So is the conclusion that we leave it like it is right now?

Reply:
Yes, but I agree with @blu3beri's suggestion to use JsonEncoder.fromString() instead of JSON.parse

Reply:
Implemented!